### PR TITLE
Fix styles for LayerControlsContainer and Console on Qt6

### LINKED ISSUE #5529: Qt6: visual error: non-theme border around LayerControlsContainer and Console (maybe macOS only?)
## 🐛 Bug

Using Qt6 backends there are borders around the LayerControls dock widget and the Console dock widget (when open). If the napari theme matches the OS theme, the border is very subtle, so you may need to toggle theme to see. 
Dark theme (OS light theme)
image
Light theme (OS light theme)
image

It's the padding pixels that use color from underneath the widgets. If one undocks the widgets, the issue disappears. Also during undocking, once removed from the dock, one can briefly see that the area beneath the widget is the same color as the problem border, before the theme color kicks in. 

This was originally reported in the Qt tracking issue:
https://github.com/napari/napari/issues/3880

## To Reproduce

1. in a fresh env, install pyqt6 or pyside6
2. `pip install napari` (no [all] to use the Qt6 backend)
3. open the viewer and open the console

## Expected behavior

The border should remain consistent with the napari theme.

## Environment

PySide6, arm64 macOS 13.2, napari `main` (post Qt6 fix PRs)

## Additional context

This may be macOS specific.

### PR DESCRIPTION
**Pull request by aganders3**
# Description
This fixes a couple visual bugs on macOS when using Qt6. I tested as well on Qt5 and don't notice any issues. Unfortunately I don't have simple access to many platforms to do more extensive testing.

I appreciate review from anyone with more Qt experience on whether these fixes make sense. 

## Qt6 main
Screenshot 2023-02-02 at 4 16 36 PM

## Qt5 main
Screenshot 2023-02-02 at 4 30 12 PM

## Qt6 this branch (PyQt)

Screenshot 2023-02-04 at 2 12 59 PM
Screenshot 2023-02-04 at 2 13 02 PM

### Previous iterations from this branch
Screenshot 2023-02-02 at 4 14 37 PM

Screenshot 2023-02-02 at 4 23 15 PM

Screenshot 2023-02-02 at 4 14 41 PM

## Type of change
- [x] Bug-fix (non-breaking change which fixes an issue)

# References
#3880
Closes #5529

# How has this been tested?
- [x] Tested changes with PyQt5 and PyQt6, PySide6 (macOS) and PyQt5 (linux)

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality lo

### CODE REVIEW

**Review by Czaki — APPROVED** [review]
Looks ok for me. 

@psobolewskiPhD could you check? As it was recognised as macOS specific problem

**Comment by psobolewskiPhD** [comment]
Wowza! 
I was going to make a dedicated issue for this and had started to try some qss to fix and got something similar as you. 

But....
If you look carefully, it's not *quite* the same as live/Qt5.
It's most obvious with the scrollbar on the right.

I think the issue is that on Qt5 the margin area shows below it the background window with `background` of that window, which is what we want. But in Qt6 that background window has a  different background that I've not yet figured out how to change. If you try to undock the layers or console you can see what's behind it.

Your fix is an improvement, but it's expanding the background of the widget—it does look a lot better tho!

Pinging @dalthviz who is a bit of a QSS 🧙‍♂️ I think.

**Comment by aganders3** [comment]
> But....
If you look carefully, it's not quite the same as live/Qt5.
It's most obvious with the scrollbar on the right.

Ahh I see what you mean - in the console? Hmm...

I think the LayerControlsContainer is correct and actually feels more "semantic" to me in the qss.

> I think the issue is that on Qt5 the margin area shows below it the background window with background of that window, which is what we want. But in Qt6 that background window has a different background that I've not yet figured out how to change. If you try to undock the layers or console you can see what's behind it.
 
Yes I agree this seems to be the issue, and would be nice to _also_ fix this even if we find other fixes in the process. Hopefully @dalthviz has some ideas!

**Comment by psobolewskiPhD** [comment]
Also, what makes this tricky is of you switch to dark mode in your macOS then the issue almost goes away—but not quite, it's super subtle. 
So the color bleeding through on macOS is from system theme, which if you use Night mode becomes warmer and makes it more obvious.

**Comment by aganders3** [comment]
Thanks - that's a good hint! Maybe we _should_ open that issue to discuss this 😅.

**Comment by psobolewskiPhD** [comment]
OK, i closed the big tracking issue, because I think it's easier to report and fix stuff individually.
And I made an issue for the visual:
https://github.com/napari/napari/issues/5529

I think your fix for the LayerControls is perfect. I'd have no issue merging just that, if we don't come up with something for the console, because the LayerControls are always visible, while the console is (in theory) an optional plugin.

File: napari/_qt/qt_resources/styles/02_custom.qss [hunk 1]
```diff
@@ -272,18 +272,12 @@ QFrame#empty_controls_widget {
...
-    margin: 10px;
-    margin-left: 10px;
-    margin-right: 8px;
-    margin-bottom: 4px;
-}
-
 QtLayerControlsContainer > QFrame {
-  padding: 5px;
-  padding-right: 8px;
+  padding: 5;
```
Comment (psobolewskiPhD) [inline napari/_qt/qt_resources/styles/02_custom.qss]:
```suggestion
  padding: 5px;
```

**Comment by psobolewskiPhD** [comment]
I think the console fix may involve adding a QFrame for the actual console widget? 🤔 
So that might be a napari-console PR?

**Comment by aganders3** [comment]
~~My latest commit brings the console at least closer to what we see on main:~~
Edit: never mind - this loses the margin when floating. Hmmm.

**Comment by aganders3** [comment]
Here's one more attempt/tweak for the console. I was hoping `setAttribute(Qt.WidgetAttribute.WA_StyledBackground)` could also be used to fix the color "behind" the dock widgets but I can't figure that out (yet).
Screenshot 2023-02-04 at 2 12 59 PM
Screenshot 2023-02-04 at 2 13 02 PM

**Comment by psobolewskiPhD** [comment]
On my end with PySide6 everything looks and behaves correct, docked and undocked!
I think you're officially co-QSS 🧙‍♂️!
🎉 

Edit: can you please edit the OP to include closing https://github.com/napari/napari/issues/5529

**Comment by aganders3** [comment]
Awesome! I appreciate anyone else testing on other platforms/backends.

**Comment by dalthviz** [comment]
Hi, just in case, on Windows seems like things look good with this PR for all the bindings I was able to test :+1: :

### PyQt5/Qt5 5.15.7

imagen

imagen

### PySide2/Qt5 5.15.2

imagen

imagen

### PyQt6/Qt6 6.4.2

imagen

imagen

### PySide6/Qt6 6.4.2

imagen

imagen

Also, seems like on Windows the mainwindow background (or the one shown when you are redocking a dockwidget) is not coupled with the OS theme. So even if you select the OS light theme, the background when selecting the area to dock has the application background color. Not totally sure how that can be changed, maybe some attribute or window flag as `Qt.WidgetAttribute.WA_StyledBackground` applied to the mainwindow object could help 🤔 ?

**Review by psobolewskiPhD — APPROVED** [review]

**Comment by aganders3** [comment]
Thanks for the reviews! Before merging I suggest taking a look at yet-another Qt6 PR I have in #5533. I don't think there's any harm in including both but that PR does achieve the same result with a smaller change.

**Comment by psobolewskiPhD** [comment]
I just merged https://github.com/napari/napari/pull/5533
which we had agreed is a better alternative to this, so will close.

**Pull request closed without merging**